geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/AutoConnectionSourceDUnitTest.java

```diff
@@ -199,19 +212,31 @@ public void testClientDynamicallyDropsStoppedLocator() throws Exception {
 
     // stop one of the locators and ensure that the client can find and use a server
     locator0VM.invoke("Stop Locator", this::stopLocator);
+    clientVM.invoke(() -> {
+      Pool pool = PoolManager.find(POOL_NAME);
+      await().until(() -> pool.getOnlineLocators().size() == 1);
+    });
 
-    serverVM.invoke("Start BridgeServer",
-        () -> startBridgeServer(null, getLocatorString(hostName, locator1Port)));
 
-    putAndWaitForSuccess(clientVM, REGION_NAME, "key", "value");
-    Assert.assertEquals("value", getInVM(serverVM, "key"));
+
+    int serverPort = serverVM.invoke("Start BridgeServer", () -> {
+      try {
```

Comment:
Recommend deleting the try-catch block entirely. Add `throws IOException` to the test method signature, and then you won't need the assertion on line 229 either.